src/parameterizations/lateral/MOM_MEKE.F90

```diff
@@ -5,27 +5,42 @@ module MOM_MEKE
 
 ! This file is part of MOM6. See LICENSE.md for the license.
 
-use MOM_debugging,     only : hchksum, uvchksum
-use MOM_cpu_clock,     only : cpu_clock_id, cpu_clock_begin, cpu_clock_end, CLOCK_ROUTINE
-use MOM_diag_mediator, only : post_data, register_diag_field, safe_alloc_ptr
-use MOM_diag_mediator, only : diag_ctrl, time_type
-use MOM_domains,       only : create_group_pass, do_group_pass, group_pass_type
-use MOM_error_handler, only : MOM_error, FATAL, WARNING, NOTE, MOM_mesg
-use MOM_file_parser,   only : read_param, get_param, log_version, param_file_type
-use MOM_grid,          only : ocean_grid_type
-use MOM_hor_index,     only : hor_index_type
-use MOM_restart,       only : MOM_restart_CS, register_restart_field, query_initialized
-use MOM_unit_scaling,  only : unit_scale_type
-use MOM_variables,     only : vertvisc_type
-use MOM_verticalGrid,  only : verticalGrid_type
-use MOM_MEKE_types,    only : MEKE_type
+use MOM_debugging,        only : hchksum, uvchksum
+use MOM_cpu_clock,        only : cpu_clock_id, cpu_clock_begin, cpu_clock_end, CLOCK_ROUTINE
+use MOM_diag_mediator,    only : post_data, register_diag_field, safe_alloc_ptr
+use MOM_diag_mediator,    only : diag_ctrl, time_type
+use MOM_domains,          only : create_group_pass, do_group_pass, group_pass_type
+use MOM_domains,          only : pass_vector
+use MOM_error_handler,    only : MOM_error, FATAL, WARNING, NOTE, MOM_mesg
+use MOM_file_parser,      only : read_param, get_param, log_version, param_file_type
+use MOM_grid,             only : ocean_grid_type
+use MOM_hor_index,        only : hor_index_type
+use MOM_io,               only : vardesc, var_desc, slasher
+use MOM_smartredis_MEKE,  only : smartredis_meke_CS_type, smartredis_meke_init, infer_meke
+use MOM_restart,          only : MOM_restart_CS, register_restart_field, query_initialized
+use MOM_smartredis,       only : client_type, smartredis_CS_type
+use MOM_string_functions, only : lowercase
+use MOM_unit_scaling,     only : unit_scale_type
+use MOM_variables,        only : vertvisc_type, thermo_var_ptrs
+use MOM_verticalGrid,     only : verticalGrid_type
+use MOM_MEKE_types,       only : MEKE_type
+
+use time_interp_external_mod, only : init_external_field, time_interp_external
```

Comment:
Please use MOM_interpolate rather that time_interp_external_mod. We no longer allow direct calls to the infrastructure from src code, but instead require all infrastructure calls to pass through the modules in src/framework.